### PR TITLE
MINIFICPP-2178 - Move extension tests to their directories

### PR DESCRIPTION
**Pull request by adamdebreceni**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by martinzink — APPROVED** [review]

**Comment by martinzink** [comment]
Due to merge conflicts the CI didnt run on this, could you rebase the PR?

**Review by szaszm — APPROVED** [review]

**Comment by lordgamez** [comment]
The `bugprone-unchecked-optional-access` issues can be ignored, because they will be removed, but the other clang-tidy issues should be fixed before merging:

```
2023-09-26T13:33:07.1338215Z /home/runner/work/nifi-minifi-cpp/nifi-minifi-cpp/extensions/rocksdb-repos/tests/DBProvenanceRepositoryTests.cpp:28:1: error: replace macro with enum [modernize-macro-to-enum,-warnings-as-errors]
2023-09-26T13:33:07.1338918Z #define TEST_PROVENANCE_STORAGE_SIZE (1024*100)  // 100 KB
2023-09-26T13:33:07.1349887Z ^~~~~~~~
2023-09-26T13:33:07.1350121Z                                      =         ,
2023-09-26T13:33:07.1350987Z /home/runner/work/nifi-minifi-cpp/nifi-minifi-cpp/extensions/rocksdb-repos/tests/DBProvenanceRepositoryTests.cpp:28:9: error: macro 'TEST_PROVENANCE_STORAGE_SIZE' defines an integral constant; prefer an enum instead [modernize-macro-to-enum,-warnings-as-errors]
2023-09-26T13:33:07.1352068Z #define TEST_PROVENANCE_STORAGE_SIZE (1024*100)  // 100 KB
2023-09-26T13:33:07.1352312Z         ^
2023-09-26T13:33:07.1353068Z /home/runner/work/nifi-minifi-cpp/nifi-minifi-cpp/extensions/rocksdb-repos/tests/DBProvenanceRepositoryTests.cpp:29:9: error: macro 'TEST_MAX_PROVENANCE_STORAGE_SIZE' defines an integral constant; prefer an enum instead [modernize-macro-to-enum,-warnings-as-errors]
2023-09-26T13:33:07.1390007Z #define TEST_MAX_PROVENANCE_STORAGE_SIZE (100*1024*1024)  // 100 MB
2023-09-26T13:33:07.1390384Z         ^
```

**Comment by szaszm** [comment]
@lordgamez could you confirm that it's good now?

**Comment by lordgamez** [comment]
> @lordgamez could you confirm that it's good now?

After the rebase some new clang tidy checks were introduced that should be addressed as well. There are only a few and they are simple enough that they can be fixed here IMO.

File: extensions/libarchive/tests/ManipulateArchiveTests.cpp [hunk 1]
```diff
@@ -31,13 +31,13 @@
...
+#include "unit/ProvenanceTestHelper.h"
 #include "repository/VolatileContentRepository.h"
 
-const char TEST_ARCHIVE_NAME[] = "manipulate_test_archive.tar";
+const char* TEST_ARCHIVE_NAME = "manipulate_test_archive.tar";
 const int NUM_FILES = 3;
-const char* FILE_NAMES[NUM_FILES] = {"first", "middle", "last"};
-const char* FILE_CONTENT[NUM_FILES] = {"Test file 1\n", "Test file 2\n", "Test file 3\n"};
+std::array<const char*, NUM_FILES> FILE_NAMES = {"first", "middle", "last"};
+std::array<const char*, NUM_FILES> FILE_CONTENT = {"Test file 1\n", "Test file 2\n", "Test file 3\n"};
```
Comment (szaszm) [inline extensions/libarchive/tests/ManipulateArchiveTests.cpp:40]: These should be top-level `const` IMO, or `constexpr` if possible.

File: extensions/libarchive/tests/ManipulateArchiveTests.cpp [hunk 2]
```diff
@@ -31,13 +31,13 @@
 #include "ManipulateArchive.h"
 #include "FocusArchiveEntry.h"
 #include "UnfocusArchiveEntry.h"
-#include "../unit/ProvenanceTestHelper.h"
+#include "unit/ProvenanceTestHelper.h"
 #include "repository/VolatileContentRepository.h"
 
-const char TEST_ARCHIVE_NAME[] = "manipulate_test_archive.tar";
+const char* TEST_ARCHIVE_NAME = "manipulate_test_archive.tar";
```
Comment (szaszm) [inline extensions/libarchive/tests/ManipulateArchiveTests.cpp:37]:
```suggestion
constexpr const char* TEST_ARCHIVE_NAME = "manipulate_test_archive.tar";
```

File: extensions/libarchive/tests/ManipulateArchiveTests.cpp (hunk 1 above)
Comment (szaszm) [inline extensions/libarchive/tests/ManipulateArchiveTests.cpp:40]:
```suggestion
constexpr std::array<const char*, NUM_FILES> FILE_NAMES = {"first", "middle", "last"};
constexpr std::array<const char*, NUM_FILES> FILE_CONTENT = {"Test file 1\n", "Test file 2\n", "Test file 3\n"};
```

**Review by szaszm — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**